neqo-common/src/datagram.rs

```diff
@@ -9,23 +9,20 @@ use std::{net::SocketAddr, ops::Deref};
 use crate::{hex_with_len, IpTos};
 
 #[derive(Clone, PartialEq, Eq)]
-pub struct Datagram {
+pub struct Datagram<D = Vec<u8>> {
     src: SocketAddr,
     dst: SocketAddr,
+    /// The size of each segment within the [`Datagram`]. All segments, but the
+    /// last, have the same size. The last segment can be shorter than
+    /// [`Datagram::segment_size`].
+    segment_size: usize,
```

Comment:
Is `Datagram` still the best descriptive name for this, given that it's now a sequence of datagrams of the same size? It sounds odd to me to see a datagram described as something that has segments in it?

Maybe `DatagramVec` or something like this would be better?

(Also wonder if `segment_size` -> `len` would then make sense. It's shorter.)

Reply:
Addressed in https://github.com/mozilla/neqo/pull/2093#issuecomment-2372100294.

Reply:
I'm not happy with that choice.  This is not a single Datagram any more.  It is a collection of datagrams with shared addressing (and TOS) information.  `DatagramVec` isn't entirely right (as that implies ownership and other `Vec` capabilities), but `DatagramList` might be OK.  

It's OK to use segments for the individual datagrams, but I would instead use `slice_size` for that, because it is the size of each slice.  The implication here is that the data payload is sliced up to get individual datagrams.

I also find this API to be pretty suboptimal for other reasons.  Why, when you receive multiple datagrams, would they all be the same size?